### ISSUE TITLE
Synchronize naming scheme of data and abstraction layers

### ISSUE THREAD

**Issue by haesleinhuepf** [clEsperanto/clesperantoj_prototype#7]
We're building similar structures in [pyclesperanto](https://github.com/clEsperanto/pyclesperanto) (P) and [clesperantoj](https://github.com/clEsperanto/clesperantoj_prototype) (J). Thus, we should make sure that things have similar names and potentially structure repositories similarily. This was done partly from the very beginning (thanks!) and has potential for sychronizing.

The following listing is from the inside to the outside. 

* CLIc: P and J have this in common, it's the C++ core.
* C++ layer code/folder between the CLIC and P/J.
  * P: [wrapper](https://github.com/clEsperanto/pyclesperanto/tree/main/wrapper)
  * J: [native](https://github.com/clEsperanto/clesperantoj_prototype/tree/main/native)
* C++ headers of the wrapper
  * P: [pyclesperanto.hpp](https://github.com/clEsperanto/pyclesperanto/blob/main/wrapper/pyclesperanto.hpp)
  * J: [clesperanto.hpp](https://github.com/clEsperanto/clesperantoj_prototype/blob/main/native/clesperantoj/include/clesperantoj.h)
* Auto-generated code from the native wrapper:
  * P: [_pyclic](https://github.com/clEsperanto/pyclesperanto/blob/main/pyclesperanto/__init__.py#L2)
  * J: [clesperantojWrapper](https://github.com/clEsperanto/clesperantoj_prototype/blob/main/src/gen/java/net/clesperanto/clicwrapper/clesperantojWrapper.java)
* The "image"/"buffer"/"array" data object representing the image on the GPU
  Naming this is important, because we will see it a billion times future scripts that use clesperanto
  * P: [data](https://github.com/clEsperanto/pyclesperanto/blob/main/pyclesperanto/_types.py#L8)
  * J: [ObjectJ](https://github.com/clEsperanto/clesperantoj_prototype/blob/d61ad99b840ce113ef1a3a5d1ed353b904ea0bf8/src/gen/java/net/clesperanto/clicwrapper/clesperantojWrapper.java#L26)
* The package/module
  This one is tricky. We are limited here by Java/Maven and Python/Pypi conventions. I can only upload to maven-central if the package starts with "net.clesperanto" because I'm owner of the domain. Packages on pypi typically have the same package name as the python module.
  * P: [pyclesperanto](https://github.com/clEsperanto/pyclesperanto/tree/main/pyclesperanto)
  * J: [net.clesperanto](https://github.com/clEsperanto/clesperantoj_prototype/tree/main/src/main/java/net/clesperanto)
* The clesperanto gateway
  * P: [Clesperanto](https://github.com/clEsperanto/pyclesperanto/blob/main/pyclesperanto/_clesperanto_gateway.py#L4)
  * J: [ClesperantoJ](https://github.com/clEsperanto/clesperantoj_prototype/blob/main/src/main/java/net/clesperanto/ClesperantoJ.java)

I would love to hear your optinions @StRigaud and @bnorthan 🙂

**Comment by StRigaud** [comment]
> The "image"/"buffer"/"array" data object representing the image on the GPU
> Naming this is important, because we will see it a billion times future scripts that use clesperanto
>* P: [data](https://github.com/clEsperanto/pyclesperanto/blob/main/pyclesperanto/_types.py#L8)
>* J: [ObjectJ](https://github.com/clEsperanto/clesperantoj_prototype/blob/d61ad99b840ce113ef1a3a5d1ed353b904ea0bf8/src/gen/java/net/clesperanto/clicwrapper/clesperantojWrapper.java#L26)

This one is a tricky thing as, depending on the language, the name may be a keyword or close to it. Ideally we would want something that is common to the three language and that do not create conflict. 

> The clesperanto gateway
>* P: [Clesperanto](https://github.com/clEsperanto/pyclesperanto/blob/main/pyclesperanto/_clesperanto_gateway.py#L4)
>* J: [ClesperantoJ](https://github.com/clEsperanto/clesperantoj_prototype/blob/main/src/main/java/net/clesperanto/ClesperantoJ.java)

Not so much important but if all could be `Clesperanto` ? 

For the rest, there is little to no impact to the users. We just have to keep a certain coherence (and maybe a doc?) and we should be fine I would say.

**Comment by haesleinhuepf** [comment]
> Ideally we would want something that is common to the three language and that do not create conflict.

Yes, I'm thinking of something like `Image` or `Array`, potentially with a prefix or suffix `CL` or `GPU` or ...

**Comment by haesleinhuepf** [comment]
"Image" is a great idea actually 🌞
https://github.com/clEsperanto/pyclesperanto_prototype/blob/de6ebd5e456f936b3d4476d2a839c12cc8234927/pyclesperanto_prototype/_tier0/_types.py#L6
